### ISSUE TITLE
Demonstration projects with Vue and React

### ISSUE THREAD

**Issue by leventozen** [Trendyol/baklava#240]
In this issue, we will create demonstration projects with real-world examples and use cases and how to use baklava. 

- [ ] Demonstration project with Vue
- [ ] Demonstration project with React

**Comment by muratcorlu** [comment]
We are considering using those projects also as automated integration tests, right? If so, let's also add this.

Also, I want to mention that if we include those projects inside the current repository, we should be extra cautious to not affect the developer experience of the people who don't care about those frameworks. For example, we may consider making them with their own package.json files, so a regular clone & start flow for baklava would not need to install vue, react (or angular in the future).

**Comment by leventozen** [comment]
Yes, like we decided before. I will elaborate issue before prioritizing it.

**Comment by muratcorlu** [comment]
I think we need to prioritise this (especially an integration test for React) We noticed that our react generator has a problem but we couldn't notice it since we are not checking React integration in every release. #331

**Comment by muratcorlu** [comment]
Issue title says "demonstration" but I find "integration tests" as a more urgent need. But maybe both can be done in the same solution. Our integration test projects can be used also for demonstrating them.

**Comment by muratcorlu** [comment]
I created [a separate issue](#337) for integration tests. I think a demonstration project still has good value for documentation purposes. So separating them makes sense, I think.
